### PR TITLE
E2e test restore backup

### PR DESCRIPTION
**Pull request by k15r**
**Description**

Introduce a starting point for e2e testing of the ark backup procedure. This PR includes only the go-part

**Related issue(s)**
#2271

### CODE REVIEW

File: tests/end-to-end/backup-restore-test/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,27 @@
+FROM golang:1.9-alpine3.7 as builder
```
Comment (rakesh-garimella) [inline tests/end-to-end/backup-restore-test/Dockerfile]: Please use `alpine` image here. As used here: https://github.com/kyma-project/kyma/blob/master/tests/kubeless-integration/Dockerfile

File: tests/end-to-end/backup-restore-test/restore/function.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+---
+apiVersion: kubeless.io/v1beta1
+kind: Function
+metadata:
+  name: hello
+spec:
+  handler: handler.hello
+  runtime: nodejs6
```
Comment (rakesh-garimella) [inline tests/end-to-end/backup-restore-test/restore/function.yaml]: quite small nodejs8 please :)

File: tests/end-to-end/backup-restore-test/deploy/chart/backup-test/values.yaml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+containerRegistry:
+  path: docker.io/k15r
```
Comment (rakesh-garimella) [inline tests/end-to-end/backup-restore-test/deploy/chart/backup-test/values.yaml]: why do you use your docker. I guess we can put gcr here.

File: tests/end-to-end/backup-restore-test/backup-restore.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
+# Backup and Restore E2E Tests
```
Comment (bszwarc) [inline tests/end-to-end/backup-restore-test/backup-restore.md]:
```suggestion
# Backup and restore E2E tests
```

File: tests/end-to-end/backup-restore-test/backup-restore.md [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+## Overview
+
+This project contains the end to end tests that it runs as part of the Kyma on Google Cloud Platform. The tests is written in Go.
```
Comment (bszwarc) [inline tests/end-to-end/backup-restore-test/backup-restore.md]:
```suggestion
This project contains end-to-end tests that it runs as part of the  Kyma on Google Cloud Platform installation. The tests are written in Go. 
```

File: tests/end-to-end/backup-restore-test/backup-restore.md [hunk 6]
```diff
@@ -0,0 +1,13 @@
+# Backup and Restore E2E Tests
+
+## Overview
+
+This project contains the end to end tests that it runs as part of the Kyma on Google Cloud Platform. The tests is written in Go. 
+
```
Comment (bszwarc) [inline tests/end-to-end/backup-restore-test/backup-restore.md]: Maybe we can add an introductory sentence here, like:
```
The end-to-end test scenario looks as follows:

```

File: tests/end-to-end/backup-restore-test/backup-restore.md [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+- Create resources on Kyma Cluster (e.g. functions).
```
Comment (bszwarc) [inline tests/end-to-end/backup-restore-test/backup-restore.md]:
```suggestion
- Create resources, such as functions on the Kyma cluster.
```

File: tests/end-to-end/backup-restore-test/backup-restore.md [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+- Backup Cluster.
```
Comment (bszwarc) [inline tests/end-to-end/backup-restore-test/backup-restore.md]:
```suggestion
- Backup the cluster.
```

File: tests/end-to-end/backup-restore-test/backup-restore.md [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+- Recreate Cluster.
```
Comment (bszwarc) [inline tests/end-to-end/backup-restore-test/backup-restore.md]:
```suggestion
- Recreate the cluster.
```

File: tests/end-to-end/backup-restore-test/backup-restore.md [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+- Restore Data and check if the restored resources are working like expected.
```
Comment (bszwarc) [inline tests/end-to-end/backup-restore-test/backup-restore.md]:
```suggestion
- Restore data and check if the restored resources work as expected.
```

File: tests/end-to-end/backup-restore-test/backup-restore.md [hunk 11]
```diff
@@ -0,0 +1,13 @@
...
+
+## Usage
```
Comment (bszwarc) [inline tests/end-to-end/backup-restore-test/backup-restore.md]: Will there be something under this section?

File: tests/end-to-end/backup-restore-test/deploy/chart/backup-test/values.yaml (hunk 3 above)
Comment (k15r) [reply tests/end-to-end/backup-restore-test/deploy/chart/backup-test/values.yaml]: i agree, that was just for testing until i have an image created by our system

File: tests/end-to-end/backup-restore-test/backup-restore.md (hunk 6 above)
Comment (k15r) [reply tests/end-to-end/backup-restore-test/backup-restore.md]: I updated and changed the documentation a bit. Could you take a look again?

File: tests/end-to-end/backup-restore-test/backup-restore.md [hunk 12]
```diff
@@ -0,0 +1,28 @@
+# Backup and restore E2E tests
+
+## Overview
+
+This project contains end-to-end tests that it runs as part of the  Kyma on Google Cloud Platform installation. The tests are written in Go.
```
Comment (bszwarc) [inline tests/end-to-end/backup-restore-test/backup-restore.md]:
```suggestion
This project contains end-to-end tests run during Kyma installation on Google Cloud Platform. The tests are written in Go. 
```

File: tests/end-to-end/backup-restore-test/backup-restore.md [hunk 13]
```diff
@@ -0,0 +1,28 @@
+# Backup and restore E2E tests
+
+## Overview
+
+This project contains end-to-end tests that it runs as part of the  Kyma on Google Cloud Platform installation. The tests are written in Go. 
+
+The end-to-end test scenario looks as follows:
+
+- Create a new namespace
```
Comment (bszwarc) [inline tests/end-to-end/backup-restore-test/backup-restore.md]:
```suggestion
- Create a new Namespace.
```

File: tests/end-to-end/backup-restore-test/backup-restore.md [hunk 14]
```diff
@@ -0,0 +1,28 @@
...
+- Create a new function in the namespace
```
Comment (bszwarc) [inline tests/end-to-end/backup-restore-test/backup-restore.md]:
```suggestion
- Create a new function in the Namespace.
```

File: tests/end-to-end/backup-restore-test/backup-restore.md [hunk 15]
```diff
@@ -0,0 +1,28 @@
...
+- Verify that the function works
```
Comment (bszwarc) [inline tests/end-to-end/backup-restore-test/backup-restore.md]:
```suggestion
- Verify that the function works.
```

File: tests/end-to-end/backup-restore-test/backup-restore.md [hunk 16]
```diff
@@ -0,0 +1,28 @@
...
+
+- Backup the namespace and all the resources in the namespace
```
Comment (bszwarc) [inline tests/end-to-end/backup-restore-test/backup-restore.md]:
```suggestion
- Back up the Namespace and all the resources in it.
```

File: tests/end-to-end/backup-restore-test/backup-restore.md [hunk 17]
```diff
@@ -0,0 +1,28 @@
...
+- Remove the namespace and all its components from the cluster
```
Comment (bszwarc) [inline tests/end-to-end/backup-restore-test/backup-restore.md]:
```suggestion
- Remove the Namespace and all its components from the cluster.
```

File: tests/end-to-end/backup-restore-test/backup-restore.md [hunk 18]
```diff
@@ -0,0 +1,28 @@
...
+
+- Restore the namespace and its content from the backup
```
Comment (bszwarc) [inline tests/end-to-end/backup-restore-test/backup-restore.md]:
```suggestion
- Restore the Namespace and its content from the backup.
```

File: tests/end-to-end/backup-restore-test/backup-restore.md [hunk 19]
```diff
@@ -0,0 +1,28 @@
...
+- Verify the function contained in the restored namespace works
```
Comment (bszwarc) [inline tests/end-to-end/backup-restore-test/backup-restore.md]:
```suggestion
- Verify if the function contained in the restored Namespace works.
```

File: tests/end-to-end/backup-restore-test/backup-restore.md [hunk 20]
```diff
@@ -0,0 +1,28 @@
...
+
+## Usage
+
+In order to run this end-to-end test just run
```
Comment (bszwarc) [inline tests/end-to-end/backup-restore-test/backup-restore.md]:
```suggestion
Use the following command to run the test:

```

File: tests/end-to-end/backup-restore-test/backup-restore.md [hunk 21]
```diff
@@ -0,0 +1,28 @@
...
+
+```
+helm install deploy/chart/backup-test --namespace end-to-end --name backup-test
+helm test backup-test
+```
+
+The test creates a new namespace called `restore-test-<UUID>`. This namespace contains all resources created during the test.
```
Comment (bszwarc) [inline tests/end-to-end/backup-restore-test/backup-restore.md]:
```suggestion
The test creates a new Namespace called `restore-test-<UUID>`. This Namespace contains all resources created during the test.
```

File: tests/end-to-end/backup-restore-test/backup-restore.md (hunk 11 above)
Comment (k15r) [reply tests/end-to-end/backup-restore-test/backup-restore.md]: added by one of the previous commits

File: tests/end-to-end/backup-restore-test/Dockerfile (hunk 1 above)
Comment (k15r) [reply tests/end-to-end/backup-restore-test/Dockerfile]: Updated to the latest alpine based golang image. This image is already used across kyma multiple times.

File: tests/end-to-end/backup-restore-test/restore/k8s.go [hunk 22]
```diff
@@ -0,0 +1,94 @@
...
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os/exec"
+)
+
+func createNamespace(namespace string) string {
```
Comment (joek) [inline tests/end-to-end/backup-restore-test/restore/k8s.go]: Please add Kyma-Related lables to the namespace.
Example:
```apiVersion: v1
kind: Namespace
metadata:
  annotations:
    kyma-roles: "true"
  creationTimestamp: 2019-02-11T08:46:40Z
  labels:
    env: "true"
    istio-injection: enabled
  name: qa
  resourceVersion: "4068"
  selfLink: /api/v1/namespaces/qa
  uid: 8ca8cc3f-2dd9-11e9-adaa-42010a9c003e
spec:
  finalizers:
  - kubernetes
status:
  phase: Active```

**Comment by venturasr** [comment]
Hi @PK85 and @pbochynski can you approve this PR

**Pull request merged**